### ISSUE TITLE
skip invalid table names in SQLite indexer

### ISSUE THREAD

**Issue by alvrs** [latticexyz/mud#2156]
The SQLite indexer currently breaks when it attempts to create a table for an invalid table name (ie containing spaces or invalid characters). Instead, the indexer should skip those tables.

![image](https://github.com/latticexyz/mud/assets/89248902/76c35a9d-21ea-4830-b9ff-437d1da56b5e)

**Comment by holic** [comment]
Before we do this, we should probably add some validations elsewhere so that folks aren't surprised by this when their data isn't indexed.

**Comment by holic** [comment]
We could also work around this with some sort of translation layer for invalid chars, similar to punycode converting unicode to domain-valid chars

**Comment by alvrs** [comment]
> Before we do this, we should probably add some validations elsewhere (config, registration methods, etc.) so that folks aren't surprised by this when their data isn't indexed.

Agreed, and even before that a useful quickfix would be catching and printing the error instead of killing the process
